spec/03-plugins/24-rate-limiting/05-integration_spec.lua

```diff
@@ -0,0 +1,177 @@
+        hosts        = { "redistest1.com" },
+        upstream_url = helpers.mock_upstream_url,
+      })
+      assert(helpers.dao.plugins:insert {
+        name   = "rate-limiting",
+        api_id = api1.id,
+        config = {
+          minute              = 1,
+          policy              = "redis",
+          redis_host          = REDIS_HOST,
+          redis_port          = REDIS_PORT,
+          redis_password      = REDIS_PASSWORD,
+          redis_database      = REDIS_DATABASE,
+          fault_tolerant      = false
+        },
+      })
+
+      local api2 = assert(helpers.dao.apis:insert {
+        name         = "redistest2_com",
+        hosts        = { "redistest2.com" },
+        upstream_url = helpers.mock_upstream_url,
+      })
+      assert(helpers.dao.plugins:insert {
+        name   = "rate-limiting",
+        api_id = api2.id,
+        config = {
+          minute              = 1,
+          policy              = "redis",
+          redis_host          = REDIS_HOST,
+          redis_port          = REDIS_PORT,
+          redis_password      = REDIS_PASSWORD,
+          redis_database      = REDIS_DATABASE_SECOND,
+          fault_tolerant      = false
+        }
+      })
+      assert(helpers.start_kong({
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+      }))
+      client = helpers.proxy_client()
+    end)
+
+    it("redis conn select databases", function()
+      local red = redis:new()
+      red:set_timeout(2000)
+      local ok, err = red:connect(REDIS_HOST, REDIS_PORT)
+      if not ok then
+        error("failed to connect to Redis: " .. err)
+      end
+      if REDIS_PASSWORD and REDIS_PASSWORD ~= "" then
+        local ok, err = red:auth(REDIS_PASSWORD)
+        if not ok then
+          error("failed to connect to Redis: " .. err)
+        end
+      end
+      red:select(REDIS_DATABASE)
+      local val1, err = red:dbsize()
+      if err then
+        error("failed to call dbsize: " .. err)
+      end
+      red:select(REDIS_DATABASE_SECOND)
+      local val2, err = red:dbsize()
+      if err then
+        error("failed to call dbsize: " .. err)
+      end
+      assert.are.same(0, tonumber(val1))
+      assert.are.same(0, tonumber(val2))
+
+      local res = assert(client:send {
+        method = "GET",
+        path = "/status/200",
+        headers = {
+          ["Host"] = "redistest1.com"
+        }
+      })
+      assert.res_status(200, res)
+      ngx.sleep(SLEEP_TIME) -- Wait for async timer to increment the limit
+
+      red:select(REDIS_DATABASE)
+      local val1, err = red:dbsize()
+      if err then
+        error("failed to call dbsize: " .. err)
+      end
+      red:select(REDIS_DATABASE_SECOND)
+      local val2, err = red:dbsize()
+      if err then
+        error("failed to call dbsize: " .. err)
+      end
+      assert.are.same(1, tonumber(val1))
+      assert.are.same(0, tonumber(val2))
+
+      -- rate-limiting plugin reuse api1 redis connection
```

Comment:
Curious: did you verify that this test fails with consistency without your patch?

Reply:
I verified this test cash get fail if not patched this commit.

spec/03-plugins/24-rate-limiting/05-integration_spec.lua:164: Expected objects to be the same.
Passed in:
(number) 2
Expected:
(number) 1

Reply:
Ok, thanks for confirming :)